### DISCUSSION TITLE
Misleading Login Error Message

### DISCUSSION THREAD

**Discussion by danieleruatecnes** [keycloak/keycloak#34441] [Ideas]
When trying to log in using incorrect credentials or a blocked user (too many attempts) I get the same error message.

So every time I get this error, I have to call the GET endpoint /admin/realms/{realm}/attack-detection/brute-force/users/{user-id} to check whether the user is really locked or has simply typed in the wrong password.

Is it possible to change the error message to distinguish between the two cases? This would be useful to save time and resources.

I think I have found where this is handled, I attach a code snippet from /services/src/main/java/org/keycloak/authentication/authenticators/directgrant/ValidateUsername.java

![image](https://github.com/user-attachments/assets/8d6c934d-67ac-46ff-9aaa-8ef99894f287)

I hope it is also useful for other developers.
Daniele

**Comment by danielFesenmeyer** [comment]
Seems there's already an issue for that: https://github.com/keycloak/keycloak/issues/17014

**Comment by douglaspalmer** [comment]
I have added a PR for #17014 that will allow different messages for invalid user, temporarily lockout and permanent lockout. To make use of this overwrite the relevant messages in a custom theme. However, I would recommend that you do not use it.

Firstly, it allows an attacker to circumvent brute force protection. They can figure out the number of guesses before lockout and the lockout period and adjust their attacks accordingly.

Second, it allows an attacker to lock users out indefinitely by simply providing the number of incorrect passwords required to lockout the user and the replaying the incorrect passwords when the temporary lockout expires.
